### PR TITLE
Script apply deposit

### PR DESCRIPTION
**Pull request by bh2smith**
This check to ensure the requested deposit hasn't already been applied and reverts with warning when it has. Otherwise, it apply requested deposits.

### CODE REVIEW

File: contracts/SnappBase.sol [hunk 1]
```diff
@@ -117,6 +116,7 @@ contract SnappBase is Ownable {
 
         stateRoots.push(_newStateRoot);        
         depositHashes[slot].applied = true;
-        emit StateTransition(TransitionType.Deposit, _currStateRoot, _newStateRoot, slot);
+        emit StateTransition(TransitionType.Deposit, this.stateIndex(), _newStateRoot, slot);
```
Comment (josojo) [inline contracts/SnappBase.sol:120]: the "this" is not needed, right?

**Review by josojo — APPROVED** [review]
looks good

File: contracts/SnappBase.sol (hunk 1 above)
Comment (bh2smith) [reply contracts/SnappBase.sol:120]: State index is a public view function returnin the length of state array minus one.

Comment (bh2smith) [reply contracts/SnappBase.sol:120]: That is excellent, I have removed all uses of this.publicViewFunction

**Comment by fleupold** [comment]
Why do we need this script? Shouldn't the driver be applying the deposits for us?

**Comment by josojo** [comment]
Sure, the driver is doing it.
This was just an additional script for testing...

**Comment by bh2smith** [comment]
Yes, this was more for my purposes, but it doesn't hurt to have such a thing.

**Pull request merged**